cpp/include/cuspatial/soa_readers.hpp

```diff
@@ -16,63 +16,63 @@
 
 #pragma once
 
-typedef struct gdf_column_ gdf_column; // forward declaration
-
 namespace cuspatial {
 
 /**
  * @brief read uint32_t data from file as column
  *
  * @param[in] filename: file to read
  *
- * @return gdf_column storing the uint32_t data
+ * @return column storing the uint32_t data
  **/
-gdf_column read_uint32_soa(const char *filename);
+std::unique_ptr<cudf::column> read_uint32_soa(const char *filename);
```

Comment:
Would it be useful to add `rmm::mr::device_memory_resource* mr` parameters to functions returning columns?

Reply:
Useful and required.

Reply:
There'll be a `detail` API for that, correct? Adding that to the follow up PR.

Reply:
streams go in the detail API. MRs go in both APIs. (The MR is used to allocate memory for any output columns -- so it has to be in the public API, otherwise the user has no control over allocation).